core/src/App.html

```diff
@@ -545,6 +546,27 @@
     return path;
   };
 
+  const appendModalDataToUrl = (modalPath, modalParams) => {
+    // global setting for persistence in url .. default false
+    const modalParamName = RoutingHelpers.getModalViewParamName();
+    const params = location.search
+      ? RoutingHelpers.parseParams(location.search.slice(1))
+      : {};
+    const prevModalPath = params[modalParamName];
```

Comment:
Shouldn't you check if the param is present in location.search?  like...
```
if (!prevModalPath){
    return;
}
```

Reply:
In this case there can or cannot be an existing modal. If we would return here, we would never append to the url.